packages/esm-patient-registration-app/src/patient-registration/patient-registration-utils.ts

```diff
@@ -192,11 +192,59 @@ export function getPatientIdentifiersFromFhirPatient(patient: fhir.Patient): Arr
   });
 }
 
-export function getPhonePersonAttributeValueFromFhirPatient(patient: fhir.Patient) {
+export async function getIdentifierFieldValuesFromFhirPatient(
+  patient: fhir.Patient,
+  identifierConfig,
+): Promise<{ [identifierFieldName: string]: PatientIdentifierValue }> {
+  const identifiers: FormValues['identifiers'] = {};
+  const promises: Promise<void>[] = [];
+
+  for (const identifier of patient.identifier) {
+    for (const config of identifierConfig) {
+      if (config.fhirIdentifierSystem !== identifier.system) {
+        continue;
+      }
+
+      const url = `${restBaseUrl}/patientidentifiertype/${config.openmrsIdentifierTypeUuid}`;
+
+      promises.push(
+        openmrsFetch(url)
+          .then((response) => {
+            if (!response.data?.name) {
+              return;
+            }
+            identifiers[response.data.name] = {
```

Comment:
I think the assumptions we're making here about the OpenMRS identifiers MPI identifiers correspond to need to be documented, i.e., I'm not always sure that `preferred : false` is the correct behaviour.